### ISSUE TITLE
The thousands separator in a format string does not work

### ISSUE THREAD

**Issue by jszuminski** [highcharts/highcharts#19991]
#### Actual behavior
The [**Templating** article](https://www.highcharts.com/docs/chart-concepts/templating) includes a [demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/labels/no-decimal-places) titled as 'Thousands separator, no decimal places: `{point.y:,.0f}`' where the method does not work (the `,` is not shown).

#### Live demo with steps to reproduce
https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/labels/no-decimal-places

#### Product version
Highcharts 11

#### Affected browser(s)
All?

#### Reported
This issue has been reported on the Highcharts forum [here](https://www.highcharts.com/forum/viewtopic.php?f=9&t=51571#p189325).

**Comment by karolkolodziej** [comment]
@jakubSzuminski The separator works correctly, you only need to specify in the lang object how it should be represented.
[See in the demo.](https://jsfiddle.net/BlackLabel/nye9pqdw/)

Quote from the documentation:

> Note that even though a dot and a comma symbolizes the decimal point and the thousands separator respectively, how it is actually rendered depends on the [language settings](https://api.highcharts.com/highcharts/lang).

**Comment by jakubSzuminski** [comment]
My bad, I've overlooked this. Thanks!

**Comment by Zwiqler94** [comment]
These demos found in the docs [here](https://www.highcharts.com/docs/chart-concepts/templating) should be updated to reflect how this should be done because they currently do it incorrectly: 

Thousands separator, no decimal places: {point.y:,.0f} [[Demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/labels/no-decimal-places)]
Thousands separator, one decimal place: {point.y:,.1f} [[Demo, internationalized](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/labels/one-decimal-place)]

**Comment by karolkolodziej** [comment]
You are right @Zwiqler94, these demos could be improved- thank you for bringing this up.

**Comment by RobbieTheWagner** [comment]
I do not seem to be able to get a thousands separator to show up at all on a `map` chart in the tooltips. Unsure if that is related to this issue.
